core/dimensions.cpp

```diff
@@ -93,8 +93,6 @@ bool Dimensions::isContiguousIn(const Dimensions &parent) const {
 Dim Dimensions::label(const scipp::index i) const { return m_dims[i]; }
 
 void Dimensions::relabel(const scipp::index i, const Dim label) {
-  if (label != Dim::Invalid)
-    expectUnique(*this, label);
```

Comment:
I don't think we can remove this check here, otherwise it will break a lot of things in this class.